internal/elasticsearch/index/component_template.go

```diff
@@ -0,0 +1,294 @@
+	id, diags := client.ID(componentId)
+	if diags.HasError() {
+		return diags
+	}
+	var componentTemplate models.ComponentTemplate
+	componentTemplate.Name = componentId
+
+	if v, ok := d.GetOk("metadata"); ok {
+		metadata := make(map[string]interface{})
+		if err := json.NewDecoder(strings.NewReader(v.(string))).Decode(&metadata); err != nil {
+			return diag.FromErr(err)
+		}
+		componentTemplate.Meta = metadata
+	}
+
+	if v, ok := d.GetOk("template"); ok {
+		// only one template block allowed to be declared
+		definedTempl := v.([]interface{})[0].(map[string]interface{})
+		definedAliases := definedTempl["alias"].(*schema.Set)
+		templ := models.Template{}
+
+		aliases := make(map[string]models.IndexAlias, definedAliases.Len())
+		for _, a := range definedAliases.List() {
+			alias := a.(map[string]interface{})
+			templAlias := models.IndexAlias{}
+
+			if f, ok := alias["filter"]; ok {
+				if f.(string) != "" {
+					filterMap := make(map[string]interface{})
+					if err := json.Unmarshal([]byte(f.(string)), &filterMap); err != nil {
+						return diag.FromErr(err)
+					}
+					templAlias.Filter = filterMap
+				}
+			}
+			if ir, ok := alias["index_routing"]; ok {
+				templAlias.IndexRouting = ir.(string)
+			}
+			templAlias.IsHidden = alias["is_hidden"].(bool)
+			templAlias.IsWriteIndex = alias["is_write_index"].(bool)
+			if r, ok := alias["routing"]; ok {
+				templAlias.Routing = r.(string)
+			}
+			if sr, ok := alias["search_routing"]; ok {
+				templAlias.SearchRouting = sr.(string)
+			}
+
+			aliases[alias["name"].(string)] = templAlias
+		}
+		templ.Aliases = aliases
+
+		if mappings, ok := definedTempl["mappings"]; ok {
+			if mappings.(string) != "" {
+				maps := make(map[string]interface{})
+				if err := json.Unmarshal([]byte(mappings.(string)), &maps); err != nil {
+					return diag.FromErr(err)
+				}
+				templ.Mappings = maps
+			}
+		}
+
+		if settings, ok := definedTempl["settings"]; ok {
+			if settings.(string) != "" {
+				sets := make(map[string]interface{})
+				if err := json.Unmarshal([]byte(settings.(string)), &sets); err != nil {
+					return diag.FromErr(err)
+				}
+				templ.Settings = sets
+			}
+		}
+
+		componentTemplate.Template = &templ
+	}
+
+	if v, ok := d.GetOk("version"); ok {
+		definedVer := v.(int)
+		componentTemplate.Version = &definedVer
+	}
+
+	if diags := client.PutElasticsearchComponentTemplate(&componentTemplate); diags.HasError() {
+		return diags
+	}
+
+	d.SetId(id.String())
+	return resourceComponentTemplateRead(ctx, d, meta)
+}
+
+func resourceComponentTemplateRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var diags diag.Diagnostics
+	client, err := clients.NewApiClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	compId, diags := clients.CompositeIdFromStr(d.Id())
+	if diags.HasError() {
+		return diags
+	}
+	templateId := compId.ResourceId
+
+	tpl, diags := client.GetElasticsearchComponentTemplate(templateId)
```

Comment:
```suggestion
	tpl, diags := client.GetElasticsearchComponentTemplate(templateId)
	if tpl == nil && diags == nil {
		d.SetId("")
		return diags
	}
```

This is another part of the suggestion when we need to handle the deleted resource outside of the TF

see also how it's done in template https://github.com/elastic/terraform-provider-elasticstack/blob/68ba59435c4166b6f929345d7f506811e24f8fa8/internal/elasticsearch/index/template.go#L285-L289